### PR TITLE
Add an onboarding flow

### PR DESCRIPTION
**Pull request by balloob**
image

Adds an onboarding flow. Onboarding flow is automatically triggered on first install. It is triggered by the backend by redirecting all normal pages to the onboarding page. Once finished, the onboarding page will redirect to the main page to start auth.

Depends on backend PR https://github.com/home-assistant/home-assistant/pull/15492

It's ugly, it's simple and it's just a start 👍 We will add more steps (like setting instance name/location, doing initial discovery/config entries).

### CODE REVIEW

**Comment by andrey-git** [comment]
(didn't actually look at the code)
Why require username/password?

**Comment by balloob** [comment]
Because we're creating a user/password that will be used to login to Home Assistant. The created user will be an owner account, able to create other users that can log in.

File: src/onboarding/ha-onboarding.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+      .error {
+        color: red;
+        font-weight: bold;
+      }
+    </style>
+    <div class="layout vertical center fit">
+      <img src="/static/icons/favicon-192x192.png" height="192">
+
+      <p>Create your user account.</p>
```
Comment (awarecan) [inline src/onboarding/ha-onboarding.js]: We need warn user they cannot change password after creation, at least for now

Comment (c727) [inline src/onboarding/ha-onboarding.js]: Is this view for the admin/owner account only? Maybe we should mention it then

**Review by c727 — COMMENTED** [review]
also remove src/onboarding/hao-step-user.js

File: src/onboarding/ha-onboarding.js (hunk 1 above)
Comment (balloob) [reply src/onboarding/ha-onboarding.js]: Yes, this view is only shown once during initial setup of Home Assisatant. Think of it like setting up a new Windows or MacOS installation.

**Pull request merged**